File: trunk/u2/avanode/file/fileentry.py
__author__ = 'markus'


class FileEntry:

    def __init__(self, host_id, host, port):
        self.id = host_id
        self.host = host
        self.port = port
        self.already_contacted = False

    def get_id(self):
        """
        Returns the ID
        :return:
        """
        return self.id

    def get_host(self):
        """
        Returns the host name
        :return:
        """
        return self.host

    def get_port(self):
        """
        Returns the defined port
        :return:
        """
        return self.port

    def is_already_contacted(self):
        return self.already_contacted

    def set_already_contacted(self):
        self.already_contacted = True

    def __str__(self):
        return "<FileEntry Object:: id: "+str(self.id)+" host:"+self.host+" port:"+str(self.port)+">"

File: src/de/htwsaar/kim/ava/avanode/file/FileEntry.java
package de.htwsaar.kim.ava.avanode.file;

/**
 * Created by markus on 25.12.16.
 */
public class FileEntry {

    private int id;
    private String host;
    private int port;

    private int vectorTime = 0;
    private int vectorTimeLimit = -1;

    public FileEntry(int id, String host, int port) {
        this.id = id;
        this.host = host;
        this.port = port;
    }

    public int getVectorTime() {
        return vectorTime;
    }

    public void incVectorTime() {
        vectorTime++;
    }

    public synchronized void updateVectorTime(int vectorTime) {
        this.vectorTime = vectorTime;
    }



    public int getId() {
        return id;
    }

    public String getHost() {
        return host;
    }

    public int getPort() {
        return port;
    }


    @Override
    public String toString() {
        return "("+id+" "+host+" "+port+")";
    }
}
